src/CLI/clitree/macro/acl_macro.xml

```diff
@@ -56,18 +72,86 @@
     </PARAM>
 </MACRODEF>
 
+<MACRODEF name="TCP-END-OPTIONS">
+    <PARAM
+        name="fin"
+        help="Match on the FIN bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+    <PARAM
+        name="syn"
+        help="Match on the SYN bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+    <PARAM
+        name="rst"
+        help="Match on the RST bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+    <PARAM
+        name="psh"
+        help="Match on the PSH bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+    <PARAM
+        name="ack"
+        help="Match on the ACK bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+    <PARAM
+        name="urg"
+        help="Match on the URG bit"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        optional="true"
+        />
+</MACRODEF>
+
 <MACRODEF name="IPV4-ACL">
-    <!-- permit tcp -->
+    <!-- deny <0-255> -->
+    <PARAM
+        name="protocol"
+        help="Protocol number"
+        ptype="UINT"
+        >
+        <MACRO name="IPV4-SRC-OPTIONS" arg="" ></MACRO>
+        <MACRO name="IPV4-DEST-OPTIONS" arg="" ></MACRO>
+        <MACRO name="CMD-END-OPTIONS" arg="" ></MACRO>
+    </PARAM>
+    <!-- deny icmp -->
+    <PARAM
+        name="icmp"
+        help="ICMP address"
+        ptype="SUBCOMMAND"
+        mode="subcommand"
+        >
+        <MACRO name="IPV4-SRC-OPTIONS" arg="" ></MACRO>
+        <MACRO name="IPV4-DEST-OPTIONS" arg="" ></MACRO>
+        <MACRO name="CMD-END-OPTIONS" arg="" ></MACRO>
+    </PARAM>
+    <!-- deny tcp -->
     <PARAM
         name="tcp"
         help="TCP packets"
         ptype="SUBCOMMAND"
         mode="subcommand"
         >
         <MACRO name="IPV4-SRC-OPTIONS" arg="" ></MACRO>
```

Comment:
You can close an element immediately with a `/>` instead of  `</MACRO>`. So these lines should be like:
`<MACRO name="IPV4-SRC-OPTIONS" arg="" />`

Reply:
Ideally both of them should work.

Reply:
Yeah, just looks cleaner

Reply:
Got it @jeff-yin